### PR TITLE
fix: [CDS-59633]: Fixes logic to determine generic repository format for ssh and WinRm deployment type

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

RCA ->
Issue -> Tag field not getting displayed for Ssh and winrm deployment type when Docker is selected as repository format

This issue has has been fixed now by updating the logic to determine if the repository format is generic/docker

#### Screenshots

NA

Screenshot 2023-04-26 at 5 59 59 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by PratyushGar — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx [hunk 1]
```diff
@@ -279,14 +277,39 @@ const Content = (props: ArtifactoryRenderContent): JSX.Element => {
...
+      isSshOrWinRmGenericSelected ||
       isAzureWebAppGenericSelected ||
       isCustomDeploymentGenericSelected ||
       isTasGenericSelected ||
       repoFormat === RepositoryFormatTypes.Generic
   )
 
+  useEffect(() => {
+    let serviceRepoFormat
+    /* istanbul ignore else */
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:295]: Is this needed?
What is the issue if we do not write else ?

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 2]
```diff
@@ -416,6 +416,15 @@ export const isAzureWebAppOrSshWinrmGenericDeploymentType = (
   return false
 }
 
+export const isSshWinRmGenericDeploymentType = (deploymentType: string, repo: string | undefined): boolean => {
+  if (isSSHWinRMDeploymentType(deploymentType)) {
+    // default repository format should be Generic if none is previously selected
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/utils/stageHelpers.ts:421]: Is this told by anyone? Why are we selecting generic by default?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:295]: Yes, it prevents istanbul from counting the else branch in coverage and hence the coverage is not unnecessarily reduced

File: src/modules/70-pipeline/utils/stageHelpers.ts (hunk 2 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/utils/stageHelpers.ts:421]: Resolved on call

**Pull request merged**